src/ir/context.rs

```diff
@@ -801,9 +801,32 @@ If you encounter an error missing from this list, please file an issue or a PR!"
 
     /// Mangles a name so it doesn't conflict with any keyword.
     pub fn rust_mangle<'a>(&self, name: &'a str) -> Cow<'a, str> {
-        if name.contains("@") ||
-            name.contains("?") ||
-            name.contains("$") ||
+        let subs = [
+            ("@", "_"),
+            ("?", "_"),
+            ("$", "_"),
+            ("+", "_plus"),
+            ("&", "_bitand"),
+            ("|", "_bitor"),
+            ("^", "_bitxor"),
+            ("/", "_div"),
+            ("%", "_rem"),
+            ("<<", "shl"),
+            (">>", "shr"),
+            // Don't forget that "operator-" is both the name of the binary
+            // minus and the unary minus operator (a.k.a the negation operator).
+            ("-", "_minus"),
+            ("=", "_equal"),
+            ("[]", "_index"),
+            ("!", "_not"),
+        ];
+        let mut invalid_name = false;
```

Comment:
nit: `let invalid_name = subs.iter().any(|sub| name.contains(sub.0))`.

Reply:
Fixed in the next commit